### ISSUE TITLE
histbin should not introduce NAs for numerics that are out of range of training data

### ISSUE THREAD

**Issue by mb706** [mlr-org/mlr3pipelines#272]

**Comment by prockenschaub** [comment]
What would the desired behaviour be?
* put outliers into the highest or lowest bin (with or without warning)
* put outliers into the highest or lowest bin and expand the label (in the case of e.g. a larger outlier of `11`, `(5-10]` might become`(5-11]`)
* always have additional bins `[-Inf, min_observed)` and `(max_observed, Inf]`
* create an additional bin for the outlier (in the case of e.g. a larger outlier, either `(max_observed, outlier]` or `(max_observed, Inf]`)

All of the above could be of interest, depending on the use case. Should we define all of them, have a sensible default (or even the NA as default) and change behaviour via a parameter?

**Comment by mb706** [comment]
Changing the labels it not a possibility because Learners rely on the same labels being available during training and prediction. Putting values into *new* bins is also a problem because Learners have trouble with bins they have not seen before. There are edge cases (maybe we could turn the outliers into NAs and have special usage, maybe the bins, which are ordered bins after all, are turned into integers and Learners can actually handle those, ...) and maybe we can cover these in the future. For now, and for simplicity, we will just go with the first bullet point, expand the  lowest bin from `[lowest value, cutoff)` to `[-inf, cutoff)` and similarly with upper bin).

I think this is easily done by modifying `get_state_dt`:

Turn
```r
 bins = lapply(seq_col(dt), function(i) {
        invoke(graphics::hist, dt[[i]], plot = FALSE, .args = self$param_set$get_values(tags = "hist"))$breaks
      })
```
into something like
```r
outlier_handling = self$param_set$values$outlier_handling
 bins = lapply(seq_col(dt), function(i) {
        breaks = invoke(graphics::hist, dt[[i]], plot = FALSE, .args = self$param_set$get_values(tags = "hist"))$breaks
        breaks[1] = -Inf
        breaks[length(breaks)] = Inf
        breaks
      })
```

and write some accompanying tests.

**Comment by pfistfl** [comment]
Fixed via #372
